realistic_library_simulation/sql_files/simulation.sql

```diff
@@ -0,0 +1,292 @@
+  "Publisher", "Publication Year", "Media", "Page Count", "Languages",
+  "LC Classification", "ISBN", "Dewey Decimal", "Dewey Wording", "Barcode",
+  "Acquisition Date", "Acquisition Price"
+)
+SELECT
+  y."Title", y."Author First Name", y."Author Last Name", y."Author Website",
+  y."Publisher", y."Publication Year", y."Media", y."Page Count", y."Languages",
+  y."LC Classification", y."ISBN", y."Dewey Decimal", y."Dewey Wording",
+  y."Barcode", y."Acquisition Date", y."Acquisition Price"
+FROM
+  publications,
+  generate_items(publications)
+    AS y(
+      "id" integer, "Title" text, "Author First Name" text,
+      "Author Last Name" text, "Author Website" mathesar_types.uri,
+      "Publisher" text, "Publication Year" integer, "Media" text,
+      "Page Count" integer, "Languages" text, "LC Classification" text,
+      "ISBN" text, "Dewey Decimal" text, "Dewey Wording" text, "Barcode" text,
+      "Acquisition Date" date, "Acquisition Price" mathesar_types.mathesar_money
+    );
+
+CREATE TABLE "Authors" (
+  id SERIAL PRIMARY KEY,
+  "First Name" text,
+  "Last Name" text,
+  "Website" mathesar_types.uri
+);
+
+ALTER TABLE "Items" ADD COLUMN "Author" integer REFERENCES "Authors";
+
+WITH split_cte AS (
+  SELECT *,
+    dense_rank() OVER (
+      ORDER BY
+        "Author Last Name",
+        "Author First Name",
+        "Author Website"
+      ) AS split_id
+  FROM "Items"
+), extract_ins_cte AS (
+  INSERT INTO "Authors"
+  SELECT DISTINCT split_id, "Author First Name", "Author Last Name", "Author Website"
+  FROM split_cte RETURNING 1 AS flag
+)
+UPDATE "Items" SET "Author"=split_cte.split_id FROM split_cte
+WHERE "Items".id = split_cte.id AND exists (SELECT flag FROM extract_ins_cte);
+
+ALTER TABLE "Items"
+  DROP COLUMN "Author First Name",
+  DROP COLUMN "Author Last Name",
+  DROP COLUMN "Author Website";
+
+CREATE TABLE "Publishers" (id SERIAL PRIMARY KEY, "Name" text);
+
+ALTER TABLE "Items" ADD COLUMN "Publisher_id" integer REFERENCES "Publishers";
+
+WITH split_cte AS (
+  SELECT *, dense_rank() OVER (ORDER BY "Publisher") AS split_id
+  FROM "Items"
+), extract_ins_cte AS (
+  INSERT INTO "Publishers"
+    SELECT DISTINCT split_id, "Publisher" FROM split_cte RETURNING 1 AS flag
+)
+UPDATE "Items" SET "Publisher_id"=split_cte.split_id FROM split_cte
+WHERE "Items".id = split_cte.id AND exists (SELECT flag FROM extract_ins_cte);
+
+ALTER TABLE "Items" DROP COLUMN "Publisher";
+ALTER TABLE "Items" RENAME COLUMN "Publisher_id" TO "Publisher";
+
+
+CREATE TABLE "Books" (
+  id SERIAL PRIMARY KEY,
+  "Title" text,
+  "Publication Year" integer,
+  "Media" text,
+  "Page Count" integer,
+  "Languages" text,
+  "LC Classification" text,
+  "ISBN" text UNIQUE,
+  "Dewey Decimal" text,
+  "Dewey Wording" text,
+  "Author" integer REFERENCES "Authors",
+  "Publisher" integer REFERENCES "Publishers"
+);
+
+ALTER TABLE "Items" ADD COLUMN "Publication" integer REFERENCES "Books";
```

Comment:
Please name this column "Book" instead of "Publication" @mathemancer

Reply:
Okay, done.